test/gtest/common/test_helpers.cc

```diff
@@ -208,6 +210,89 @@ void watchdog_stop()
     pthread_mutex_destroy(&watchdog.mutex);
 }
 
+static bool test_results_cmp(const test_result_t &a, const test_result_t &b)
+{
+    return a.second > b.second;
+}
+
+void analyze_test_results()
+{
+    // GTEST_REPORT_LONGEST_TESTS=100 will report TOP-100 longest tests
+    /* coverity[tainted_data_return] */
+    char *env_p = getenv("GTEST_REPORT_LONGEST_TESTS");
+    if (env_p == NULL) {
+        return;
+    }
+
+    size_t max_name_size = 0;
+    int top_n;
+
+    if (!strcmp(env_p, "*")) {
+        top_n = std::numeric_limits<int>::max();
+    } else {
+        top_n = atoi(env_p);
+        if (!top_n) {
+            return;
+        }
+    }
+
+    ::testing::UnitTest *unit_test = ::testing::UnitTest::GetInstance();
+    std::vector<test_result_t> test_results;
+
+    if (unit_test == NULL) {
+        ADD_FAILURE() << "Unable to get the Unit Test instance";
+        return;
+    }
+
+    for (int i = 0; i < unit_test->total_test_case_count(); i++) {
+        const ::testing::TestCase *test_case = unit_test->GetTestCase(i);
+        if (test_case == NULL) {
+            ADD_FAILURE() << "Unable to get the Test Case instance with index "
+                          << i;
+            return;
+        }
+
+        for (int i = 0; i < test_case->total_test_count(); i++) {
+            const ::testing::TestInfo *test = test_case->GetTestInfo(i);
+            if (test == NULL) {
+                ADD_FAILURE() << "Unable to get the Test Info instance with index "
+                              << i;
+                return;
+            }
+
+            if (test->should_run()) {
+                const ::testing::TestResult *result = test->result();
+                std::string test_name               = test->test_case_name();
+
+                test_name += ".";
+                test_name += test->name();
+
+                test_results.push_back(std::make_pair(test_name,
+                                                      result->elapsed_time()));
+
+                max_name_size = std::max(test_name.size(), max_name_size);
+            }
+        }
+    }
+
+    std::sort(test_results.begin(), test_results.end(), test_results_cmp);
+
+    top_n = std::min((int)test_results.size(), top_n);
+    if (!top_n) {
+        return;
+    }
+
+    int max_index_size = ucs::to_string(top_n).size();
+    std::cout << std::endl << "TOP-" << top_n << " longest tests:" << std::endl;
```

Comment:
Will this string be printed only, if negative digit is provided for top_n?

Reply:
yes, you're right
do we need to add sanity checks for the value passed to `GTEST_REPORT_LONGEST_TESTS`?

Reply:
i don't think so, it's internal/test stuff

Reply:
I'd just ignore negative values (check for >0 instead of !=0), but it's too late anyway